addons/stock/models/stock_move.py

```diff
@@ -615,6 +615,46 @@ def action_show_details(self):
         """
         self.ensure_one()
 
+        if self.origin_returned_move_id and float_is_zero(self.quantity_done, precision_rounding=self.product_uom.rounding):
+            origin_quantity = self.origin_returned_move_id.product_uom._compute_quantity(self.origin_returned_move_id.product_uom_qty, self.product_uom, round=False)
+            if self.product_id.tracking == 'lot':
+                if float_compare(self.product_qty, origin_quantity, precision_rounding=self.product_uom.rounding) == 0:
+                    move_lines_commands = []
+                    for move_line in self.origin_returned_move_id.move_line_ids:
+                        move_line_vals = self._prepare_move_line_vals(quantity=0)
+                        move_line_vals['qty_done'] = move_line.product_uom_id._compute_quantity(move_line.qty_done, self.product_uom)
+                        move_line_vals['product_uom_id'] = self.product_uom.id
+                        move_line_vals['lot_id'] = move_line.lot_id.id
+                        move_lines_commands.append((0, 0, move_line_vals))
+                    self.write({'move_line_ids': move_lines_commands})
+                elif len(set(move_line.lot_id for move_line in self.origin_returned_move_id.move_line_ids)) == 1:
```

Comment:
Can be simplify `len(self.origin_returned_move_id.move_line_ids.lot_id) == 1`. The mapped discard duplicate